runtime/wasm/src/mapping.rs

```diff
@@ -138,6 +139,7 @@ pub(crate) struct MappingContext {
     pub(crate) state: BlockState,
 }
 
+// TODO: This should not be called clone().
```

Comment:
`{clone,copy}_with_empty_block_state()` perhaps? I agree, it isn't _truly_ cloning due to the block state being dropped.

Reply:
Oops, I meant to go back and rename this. The problem is that with the kind of work that I'm going it's desirable to go around and remove calls to `clone()` that were only put there to work around lifetime problems. It's a useful heuristic to assume that converting a `clone()` to a `&` does not change any runtime behavior (besides performance).

I'll go rename this now.